pulpcore/pulpcore/app/models/repository.py

```diff
@@ -221,25 +207,147 @@ class RepositoryContent(Model):
 
         content (models.ForeignKey): The associated content.
         repository (models.ForeignKey): The associated repository.
+        vadded (models.ForeignKey): The RepositoryVersion which added the referenced Content.
+        vremoved (models.ForeignKey): The RepositoryVersion which removed the referenced Content.
     """
-    content = models.ForeignKey('Content', on_delete=models.CASCADE)
+    content = models.ForeignKey('Content', on_delete=models.CASCADE,
+                                related_name='version_memberships')
     repository = models.ForeignKey(Repository, on_delete=models.CASCADE)
+    vadded = models.ForeignKey('RepositoryVersion', related_name='added_memberships')
+    vremoved = models.ForeignKey('RepositoryVersion', null=True, related_name='removed_memberships')
+
+    class Meta:
+        unique_together = (('repository', 'content', 'vadded'),
+                           ('repository', 'content', 'vremoved'))
+
+
+class RepositoryVersion(Model):
+    """
+    A version of a repository's content set.
+    Fields:
+        number (models.PositiveIntegerField): A positive integer that uniquely identifies a version
+            of a specific repository. Each new version for a repo should have this field set to
+            1 + the most recent version.
```

Comment:
Suppose the latest version is deleted and then a new version is created. It would have the same version number as the deleted version. I think a better idea would be to store a field (e.g. latest_version) on the repository.

Reply:
I see in delete_version() you prevent users from deleting the latest to prevent the problem I mention. You may want to consider the solution I mention though since it would allow users to delete the latest version.